### PR TITLE
Improve test suite

### PR DESCRIPTION
**Pull request by deeky666**
This PR adopts DBAL's latest test suite improvements from https://github.com/doctrine/dbal/pull/597, https://github.com/doctrine/dbal/pull/643, https://github.com/doctrine/dbal/pull/779 and https://github.com/doctrine/dbal/pull/758
Supersedes https://github.com/doctrine/doctrine2/pull/1285

### CODE REVIEW

File: tests/Doctrine/Tests/TestUtil.php [hunk 1]
```diff
@@ -28,92 +33,157 @@ class TestUtil
...
+
+    private static function hasRequiredConnectionParams()
+    {
+        return isset(
+        $GLOBALS['db_type'],
+        $GLOBALS['db_username'],
+        $GLOBALS['db_password'],
+        $GLOBALS['db_host'],
+        $GLOBALS['db_name'],
+        $GLOBALS['db_port']
```
Comment (stof) [inline tests/Doctrine/Tests/TestUtil.php]: wrong indentation

Comment (deeky666) [inline tests/Doctrine/Tests/TestUtil.php]: Thank you, copy-paste :D Will fix.

**Comment by Ocramius** [comment]
amaze

**Comment by TomasVotruba** [comment]
@deeky666 Great job!

**Comment by deeky666** [comment]
You're welcome. Actually I forgot to bake in https://github.com/doctrine/dbal/pull/779 (doh!). Will do now

**Comment by deeky666** [comment]
okay now it's finished!

**Pull request merged**